### PR TITLE
Use cc instead of cmake to build ittnotify lib

### PR DESCRIPTION
**Pull request by jlb6740**

### CODE REVIEW

**Comment by jlb6740** [comment]
I think this is ok but we do need to test what happens if it were to be published before we merge. Looks like there was never any windows support. This updates the linux and mac support.

**Review by abrown — APPROVED** [review]
LGTM! I re-ran the example @jlb6740 documented [here](https://docs.wasmtime.dev/examples-profiling-vtune.html#vtune-example) using these new build changes and everything still seems to work as expected. My updates to the Wasmtime documentation are at https://github.com/bytecodealliance/wasmtime/pull/3604.

File: ittapi-rs/build.rs [hunk 1]
```diff
@@ -13,25 +12,23 @@ fn main() {
...
-            .build();
-
-        println!("cargo:rustc-link-search={}/build/bin/", out_path.display());
-        println!("cargo:rustc-link-lib=static=ittnotify");
+        cc::Build::new().file("src/ittnotify/ittnotify_static.c")
+            .file("src/ittnotify/jitprofiling.c")
+            .include("src/ittnotify/")
+            .include("include/")
+            .compile("ittnotify64");
+        println!("cargo:rustc-link-lib=static=ittnotify64");
```
[the remark this answers is not in the document]
Comment (jlb6740) [reply ittapi-rs/build.rs]: Ok .. I pushed an update. Pretty sure it still works 🤷‍♀️ 😊

File: ittapi-rs/build.rs [hunk 2]
```diff
@@ -13,25 +12,23 @@ fn main() {
...
-            .no_build_target(true)
-            .build();
-
-        println!("cargo:rustc-link-search={}/build/bin/", out_path.display());
-        println!("cargo:rustc-link-lib=static=ittnotify");
+        cc::Build::new().file("src/ittnotify/ittnotify_static.c")
+            .file("src/ittnotify/jitprofiling.c")
+            .include("src/ittnotify/")
+            .include("include/")
+            .compile("ittnotify64");
```
[the remark this answers is not in the document]
Comment (jlb6740) [reply ittapi-rs/build.rs:19]: @ekovanova @abrown I think this is ready for merge +1

[the remark this answers is not in the document]
Comment (abrown) [reply ittapi-rs/build.rs:19]: Yeah, looks good to me!

[the remark this answers is not in the document]
Comment (ekovanova) [reply ittapi-rs/build.rs:19]: LGTM! 

Also I should note that ITT/JIT APIs C library was renamed - it's no longer contain "32" or "64" postfix: 
https://github.com/intel/ittapi/commit/8601f825f50b1091c2b0911ce2b9a91e7ced3e69
Probably you would like to rename it is well.

I checked the documentation change - the command line 'amplxe-cl' is deprecated (but actually still works as expected)  and we recommend to use 'vtune' instead.

[the remark this answers is not in the document]
Comment (abrown) [reply ittapi-rs/build.rs:19]: Good to know; opened #35 to fix this.

**Pull request merged**